documentation/docs/getting-started/core-concepts.md

```diff
@@ -0,0 +1,153 @@
+# Core Concepts
+
+### Middlewares
+
+A middleware in Fusion.js is similar to middlewares in HTTP servers like Express.js (in fact, they are built on top of Koa middlewares). Fusion.js middlewares are functions that run on the server when an HTTP request occurs. However, unlike Koa/Express middlewares, Fusion.js middleware functions also run on the browser on page load.
+
+### Services
+
+A service is any library that exposes a programmatic API. The main difference between a Fusion.js service and a regular library is that services are dynamically injected via a dependency injection system, whereas libraries would typically be imported statically via ES6 import statements.
+
+### Plugin
+
+A Fusion.js plugin is a container that encapsulates a service and a middleware. Plugins declare what dependencies they rely on (via the DI system) and may expose a service, a middleware, or both. Services and middlewares that are exposed by a plugin can consume services from other plugins.
+
+Fusion.js plugins are suitable for encapsulating integrations with libraries, even when they require making changes to multiple areas of concern (e.g. server, browser, React providers).
+
```

Comment:
Let's link to the plugin doc: https://fusionjs.com/api/fusion-docs/creating-a-plugin